lib/button-control/inc/button.hpp

```diff
@@ -0,0 +1,99 @@
+#include <Arduino.h>
+
+namespace ns_button
+{
+    const unsigned long debounce_delay = 75;
+    const unsigned long medium_press = 1000;
+    const unsigned long long_press = 5000;
+    
+    enum Button_state {
```

Comment:
Why not put them all and create button class ?

Reply:
Because we need static functions as callbacks to the interrupts.
I think that the buttons have to be something static.

Reply:
I looked at it and it turns out that you can use lambda expressions in `attachInterrupt()`, you only need to attach the library ` #include <FunctionalInterrupt.h> ` from esp